### PR TITLE
Add Rain-Doh black box, add Macro.tryCopier() function

### PR DESCRIPTION
**Pull request by Rinn**
- Macro.tryCopier() checks various conditions on whether or not a copier item or skill should be used
- Add support for Rain-Doh black box
- Use a spooky putty monster if it is invalid after ascending

Ran two days and it seems to be working correctly.

### CODE REVIEW

File: src/fights.ts [hunk 1]
```diff
@@ -370,6 +373,11 @@ function embezzlerSetup() {
   if (!get("_cameraUsed") && !have($item`shaking 4-d camera`)) {
     retrieveItem($item`4-d camera`);
   }
+
+  // Fix invalid spooky putty monster (caused by ascending)
```
Comment (horrible-little-slime) [inline src/fights.ts:377]: You may want to examine the spooky putty monster here, to make sure mafia knows what's in there.

Comment (Rinn) [reply src/fights.ts:377]: yeah i'll add that

File: src/combat.ts [hunk 2]
```diff
@@ -53,6 +55,67 @@ export class Macro extends LibramMacro {
     return new Macro().tryHaveSkill(skillOrName);
   }
 
+  tryCopier(itemOrSkill: Item | Skill, monster: Monster = $monster`none`): Macro {
+    switch (itemOrSkill) {
+      case $item`Spooky Putty sheet`:
+        return this.externalIf(
+          get("spookyPuttyCopiesMade") < 5 && !have($item`Spooky Putty monster`),
```
Comment (soolar) [inline src/combat.ts:62]: This should also check to make sure `spookyPuttyCopiesMade + _raindohCopiesMade < 6`, shouldn't it? And same for below with black box.

**Comment by horrible-little-slime** [comment]
Resolves #207

File: src/combat.ts [hunk 3]
```diff
@@ -53,6 +53,65 @@ export class Macro extends LibramMacro {
...
+      case $item`alpine watercolor set`:
+        return this.tryItem(itemOrSkill);
+      case $item`LOV Enamorang`:
+        return this.externalIf(
+          get("_enamorangs") < 5 && get("enamorangMonster") === $monster`none`,
+          Macro.tryItem(itemOrSkill)
+        );
+      case $skill`Digitize`:
+        return this.externalIf(
+          get("_sourceTerminalDigitizeUses") <
```
Comment (horrible-little-slime) [inline src/combat.ts:99]: This is redundant with trySkill--digitize shouldn't show up if you are out of digitizes

File: src/combat.ts [hunk 4]
```diff
@@ -53,6 +53,65 @@ export class Macro extends LibramMacro {
...
+          !get("_envyfishEggUsed") && !have($item`envyfish egg`),
+          Macro.tryItem(itemOrSkill)
+        );
+      case $item`print screen button`:
+        return this.tryItem(itemOrSkill);
+      case $item`alpine watercolor set`:
+        return this.tryItem(itemOrSkill);
+      case $item`LOV Enamorang`:
+        return this.externalIf(
+          get("_enamorangs") < 5 && get("enamorangMonster") === $monster`none`,
```
Comment (horrible-little-slime) [inline src/combat.ts:94]: enamorangMonster will never be `none`--it will be null or undefined, instead.

File: src/fights.ts [hunk 5]
```diff
@@ -379,6 +385,16 @@ function embezzlerSetup() {
...
+  if (have($item`Spooky Putty monster`) && get("spookyPuttyMonster") === $monster`none`) {
```
Comment (horrible-little-slime) [inline src/fights.ts:390]: A similar thing can happen to rain-doh if you use the copier while wearing papier-mache or other text-obfuscating conditions--is it worth adding a refresher there?

File: src/combat.ts (hunk 3 above)
Comment (Rinn) [reply src/combat.ts:99]: My thought was to reduce the size of the macro that's being submitted, but yeah you're right it's technically not necessary.

**Review by horrible-little-slime — APPROVED** [review]
I am slightly concerned about using putty and the box in the same fight for reasons discussed in discord, but for reasons discussed in discord it is unlikely to actually lead to explosions.

**Pull request merged**